modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/factory/CustomElementCETImplFactoryImpl.java

```diff
@@ -113,31 +123,43 @@ public void validate(
 			}
 			else {
 				throw new ClientExtensionEntryTypeSettingsException(
-					"html-element-name-contains-invalid-character-x", c);
+					"Custom element HTML element name contains an" +
+					"invalid character: " + c,
+					"html-element-name-contains-invalid-character-x", c
+				);
 			}
 		}
 
 		if (!containsDash) {
 			throw new ClientExtensionEntryTypeSettingsException(
-				"html-element-name-must-contain-at-least-one-hyphen");
+				"Custom element HTML element name must contain at least one " +
+				"hyphen",
+				"html-element-name-must-contain-at-least-one-hyphen"
+			);
 		}
 
 		if (_reservedHTMLElementNames.contains(htmlElementName)) {
 			throw new ClientExtensionEntryTypeSettingsException(
-				"x-is-a-reserved-html-element-name", htmlElementName);
+				"Reserved custom element HTML element name " + htmlElementName,
```

Comment:
`Reserved custom element HTML element name ` should probably include a colon `Reserved custom element HTML element name: `